packages/gatsby/src/internal-plugins/prod-404/gatsby-node.js

```diff
@@ -4,18 +4,25 @@ const { actions } = require(`../../redux/actions`)
 const PROD_404_PAGE_PATH = `/404.html`
 
 let page404 = null
-exports.onCreatePage = ({ page, store, actions }) => {
+emitter.on(`CREATE_PAGE`, action => {
```

Comment:
this drops the createPages time for 100k pages in create-pages from ~8s to ~4.5s